### ISSUE TITLE
Mcfly + zsh + fig

### ISSUE THREAD

**Issue by pitosalas** [cantino/mcfly#301]
Previously I was enjoying mcfly with zsh all day long and was happy.

Then I installed fig (also very nice) and mcfly doesn't launch when I type ^r

**Comment by navazjm** [comment]
Ah, never heard of fig until now. I just installed it and imported my dotfiles into fig and seems to run McFly perfectly for me. 

[Here is a link on how to import your dotfiles.](https://fig.io/user-manual/dotfiles/import)

In the above link, it also mentions to remove or comment out the original ~/.zshrc, ~/.zprofile, or ~/.zshenv  file to prevent things from loading twice. I did not do this and McFly still ran for me. However, it may help you.

If you have already imported your dotfiles, I would try adding a path to cargo and see if that solves your issue.

![Screenshot 2022-11-24 at 10 55 10 PM](https://user-images.githubusercontent.com/44414214/203904124-f6a38231-73a0-496e-bb9c-02f99acb7c5f.png)

Also be sure to run the following command after making changes in the fig dashboard.
```sh
fig source
```

Unfortunately, I do not think this is an issue with McFly so it may just take time tinkering with Fig to get it to work.

**Comment by pitosalas** [comment]
Thanks. You’re right, in was able to tinker a lot to get it to work. Nice app though, right?Pito Salas Computer ScienceBrandeis Univeristy On Nov 25, 2022, at 12:02 AM, Michael Navarro ***@***.***> wrote:﻿
Ah, never heard of fig until now. I just installed it and imported my dotfiles into fig and seems to run McFly perfectly for me.
Here is a link on how to import your dotfiles.
In the above link, it also mentions to remove or comment out the original ~/.zshrc, ~/.zprofile, or ~/.zshenv  file to prevent things from loading twice. I did not do this and McFly still ran for me. However, it may help you.
If you have already imported your dotfiles, I would try adding a path to cargo and see if that solves your issue.

Also be sure to run the following command after making changes in the fig dashboard.
fig source
Unfortunately, I do not think this is an issue with McFly so it may just take time tinkering with Fig to get it to work.

**Comment by pitosalas** [comment]
Hi Again,

It broke again. I know it's not mcfly's bug but maybe you can help me with a question.

Why would this happen:


~ $(mcfly init zsh)
zsh: no such file or directory: #!/bin/zsh

~ ls /bin/zsh
/bin/zsh

Something wrong with the path but I just understand how it could happen.

- Thanks,

Pito

**Comment by navazjm** [comment]
I am assuming you are manually running `$(mcfly init zsh)` inside your terminal. If so, you need to change it to `eval "$(mcfly init zsh)"` or `source <(mcfly init zsh)`.

This will allow you to run mcfly by pressing `Ctrl + r`

From `man bash` :
![Screenshot 2022-11-30 at 8 47 08 PM](https://user-images.githubusercontent.com/44414214/204954288-4b1090d3-dac8-490a-9e0d-f940a8a112a1.png)
![Screenshot 2022-11-30 at 8 48 45 PM](https://user-images.githubusercontent.com/44414214/204954293-6663a454-a500-4c79-b994-9179c0f148af.png)

**Comment by pitosalas** [comment]
Yes, you got it right! I know the difference but my fingers typed the wrong thing.

Great product!
